### PR TITLE
Flag status fix

### LINKED ISSUE #54: when phase I subproblem is infeasible
It's incorrect to return ``:Infeasible`` status. We should print an error message indicating numerical issues.

### PR DESCRIPTION
**Pull request by emreyamangil**
This PR takes an initial aim at returning proper flags, though Convex.jl is causing issues!!

Kinda Closes #54

### CODE REVIEW

File: src/conic.jl [hunk 1]
```diff
@@ -844,17 +844,26 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
     cputime_conic += time() - start_conic
 
     ini_conic_status = MathProgBase.status(ini_conic_model)
-    ini_conic_objval = MathProgBase.getobjval(ini_conic_model)
-    if ini_conic_status == :Optimal || ini_conic_status == :Suboptimal
+    if ini_conic_status == :Optimal || ini_conic_status == :SubOptimal
```
Comment (mlubin) [inline src/conic.jl]: ``:Suboptimal``

File: src/conic.jl [hunk 2]
```diff
@@ -944,11 +953,9 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
             MathProgBase.optimize!(inf_dcp_model)
             cputime_conic += time() - start_conic
 
-            inf_cut_generator = true
-            if MathProgBase.status(inf_dcp_model) == :Infeasible
-                (m.verbose > 1) && println("INF DCP Infeasible")
-                m.status = :Infeasible
-                return
+            inf_dcp_status = MathProgBase.status(inf_dcp_model)
+            if inf_dcp_status != :Optimal && inf_dcp_status != :SubOptimal
```
Comment (mlubin) [inline src/conic.jl]: ``:Suboptimal``

File: src/nonlinear.jl [hunk 3]
```diff
@@ -469,15 +468,25 @@ function MathProgBase.optimize!(m::PajaritoModel)
     start_nlp = time()
     MathProgBase.optimize!(ini_nlp_model)
     cputime_nlp += time() - start_nlp
+
     ini_nlp_status = MathProgBase.status(ini_nlp_model)
-    if ini_nlp_status == :Infeasible || ini_nlp_status == :InfeasibleOrUnbounded
+    if ini_nlp_status == :Optimal || ini_nlp_status == :SubOptimal
```
Comment (mlubin) [inline src/nonlinear.jl]: Same here

File: src/nonlinear.jl [hunk 4]
```diff
@@ -600,10 +603,9 @@ function MathProgBase.optimize!(m::PajaritoModel)
             start_nlp = time()
             MathProgBase.optimize!(inf_model)
             cputime_nlp += time() - start_nlp
-            if MathProgBase.status(inf_model) == :Infeasible
-                (m.verbose > 2) && println("INF NLP Infeasible")
-                m.status = :Infeasible
-                return
+            inf_model_status = MathProgBase.status(inf_model)
+            if inf_model_status != :Optimal && inf_model_stauts != :SubOptimal
```
Comment (mlubin) [inline src/nonlinear.jl]: Same here

**Comment by mlubin** [comment]
Fill the solution with ``NaN`` to work around the Convex.jl issue

File: test/conictest.jl [hunk 5]
```diff
@@ -46,6 +46,25 @@ end
     end
 end
 
+# TODO Enable this test after Convex.jl issue is resolved!
```
Comment (mlubin) [inline test/conictest.jl]: Out of date comment?

File: src/nonlinear.jl [hunk 6]
```diff
@@ -600,10 +606,9 @@ function MathProgBase.optimize!(m::PajaritoModel)
             start_nlp = time()
             MathProgBase.optimize!(inf_model)
             cputime_nlp += time() - start_nlp
-            if MathProgBase.status(inf_model) == :Infeasible
-                (m.verbose > 2) && println("INF NLP Infeasible")
-                m.status = :Infeasible
-                return
+            inf_model_status = MathProgBase.status(inf_model)
+            if inf_model_status != :Optimal && inf_model_stauts != :Suboptimal
+                error("NLP Solver Failure.")
```
Comment (mlubin) [inline src/nonlinear.jl]: We should print a warning an return ``:Error`` status, not crash the program with ``error()``

File: src/conic.jl [hunk 7]
```diff
@@ -844,17 +844,29 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
...
-        (m.verbose > 1) && println("Conic Relaxation Infeasible")
+    elseif ini_conic_status == :Infeasible
+        (m.verbose > 1) && println("Conic Relaxation Infeasible.")
         m.status = :Infeasible
-        return       
+        m.solution = fill(NaN, m.numVar)
+        return    
+    # TODO Figure out the details for this condition to hold!   
+    elseif ini_conic_status == :Unbounded
+        (m.verbose > 1) && println("Conic Relaxation Unbounded.")
```
Comment (mlubin) [inline src/conic.jl]: I would turn this into a ``warn`` because the ``:InfeasibleOrUnbounded`` status is confusing

Comment (emreyamangil) [inline src/conic.jl]: Yupp!!

File: src/nonlinear.jl (hunk 6 above)
Comment (emreyamangil) [inline src/nonlinear.jl]: Fixing!

**Comment by emreyamangil** [comment]
Pajarito started to look good isn't it?? :)

**Pull request merged**